### DISCUSSION TITLE
TypeError: can't convert BigInt to number with chain.id

### DISCUSSION THREAD

**Discussion by AlphaDio** [wevm/viem#1007] [Question] [answered]
![image](https://github.com/wagmi-dev/viem/assets/1067597/1451b53b-b4b4-4a62-af62-df1759fd9f43)

I keep trying to make the signTypedMessage message work, and here I get the chain id for the typed Message from Wagmi.

```
const { chain } = useNetwork()
```

```
walletClient
      .signTypedData({
        account,
        domain: {
          name: 'maker',
          **chainId: chain.id,**
          verifyingContract: makerAddress as `0x${string}`,
          version: '1',
        },
```

For me, this should work _without much fuss_ (another thing I'm starting to find really tiring about wagmi and viem, I sponsored this project once..), but I keep getting this error when doing the signing:
Uncaught (in promise) TypeError: Cannot convert a BigInt value to a number
    at Math.pow (<anonymous>)
    at c (toHex.ts:173:21)
    at e (typedData.ts:41:9)
    at Xt (typedData.ts:67:5)
    at signTypedData.ts:163:3
    at d (regeneratorRuntime.js:44:17)
    at Generator.<anonymous> (regeneratorRuntime.js:125:22)
    at Generator.next (regeneratorRuntime.js:69:21)
    at r (asyncToGenerator.js:3:20)
    at f (asyncToGenerator.js:22:9)

Can someone explain to me what is happening here? I verified and it seems stuck when trying to convert size to a BigInt at line 173.

**Comment by jxom** [comment]
What bundler/runtime are you using? Next.js, Vite, Node, etc? A minimal reproduction would help a lot.

> another thing I'm starting to find really tiring about wagmi and viem

what else do you find fussy?

**Reply by jxom** [reply]
Could be related: https://github.com/wagmi-dev/viem/issues/558#issuecomment-1558201543

**Reply by AlphaDio** [reply]
I'm using Node. What specifically would you be looking for?

I had to deal with the multicall contract being called while developing locally, that was ages ago and this is solved now, but at the time it was not documented and it made me waste days trying to pinpoint something that had nothing to do with my code. Overall **I** (might be a 'me' problem) feels like paying the price of "good ideas" while using the framework.

**Reply by AlphaDio** [reply]
Create-React-App too, unfortunately..

**Reply by jxom** [reply, marked as answer]
If you are using CRA, you can modify the browserlist: https://github.com/hirosystems/stacks.js/issues/1096#issuecomment-946350299

**Reply by AlphaDio** [reply]
It seems like it fixes the issue. Thanks for the insight 👍  At least I know it's not entirely on my hand.

**Reply by jxom** [reply]
Could be worth upgrading to a more modern bundler like Vite or Next.js. I don’t believe CRA is maintained anymore.

**Reply by AlphaDio** [reply]
Yep, it's pretty much dead, I'm working on upgrading the stack of my app, but it was already hard enough to update wagmi :P, so I guess I will wait a moment before doing so.

All jokes aside, great job on the library, still a fan, I just hope for less pain points on my side.
